src/main/scala/scalismo/image/DiscreteImageDomain.scala

```diff
@@ -0,0 +1,196 @@
+/*
+ * Copyright 2015 University of Basel, Graphics and Vision Research Group
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package scalismo.image
+
+import scalismo.common.{BoxDomain, DiscreteDomain}
+import scalismo.geometry.{
+  _1D,
+  _2D,
+  _3D,
+  EuclideanVector,
+  EuclideanVector1D,
+  EuclideanVector2D,
+  EuclideanVector3D,
+  IntVector,
+  IntVector1D,
+  IntVector2D,
+  IntVector3D,
+  NDSpace,
+  Point
+}
+
+import scala.language.implicitConversions
+
+case class DiscreteImageDomain[D: NDSpace](structuredPoints: StructuredPoints[D]) extends DiscreteDomain[D] {
+
+  override def pointSet: StructuredPoints[D] = structuredPoints
+
+  def origin = pointSet.origin
+  def spacing = pointSet.spacing
+  def size = pointSet.size
+
+  def boundingBox: BoxDomain[D] = {
+
+    // The image bounding box is 1*spacing larger than the bounding box of the point of the domain, as
+    // every point of the domain represents one voxel.
```

Comment:
might be important to mention here that this implementation implies that for each voxel, the point it defines is at the corner to the negative side of each axis. Or is this convention common sense?

Reply:
Thanks. Added the following clarification:
```
// (The voxel that is defined by each grid point extends by the length spacing(i) into the
// i-th space direction).
```